src/Mono.Android/Android.App/SyncContext.cs

```diff
@@ -14,7 +15,15 @@ public override SynchronizationContext CreateCopy ()
 
 		public override void Post (SendOrPostCallback d, object state)
 		{
-			using (Handler h = new Handler(Application.Context.MainLooper)) {
+			var looper = Application.Context?.MainLooper;
+			if (looper == null) {
```

Comment:
1. This check should be moved into a helper method.
2. The method from (1) should also be called from `Send()`.